nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/dynamodb/PutDynamoDBRecord.java

```
@@ -0,0 +1,363 @@
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final Map<String, String> attributes = new HashMap<>(flowFile.getAttributes());
+        attributes.put(DYNAMODB_CHUNKS_PROCESSED_ATTRIBUTE, String.valueOf(result.getSuccessfulChunks()));
+        final FlowFile outgoingFlowFile = session.putAllAttributes(flowFile, attributes);
+
+        if (result.isSuccess()) {
+            session.transfer(outgoingFlowFile, REL_SUCCESS);
+        } else if (result.getThrowable().getCause() instanceof ProvisionedThroughputExceededException) {
+             // When DynamoDB returns with {@code ProvisionedThroughputExceededException}, the client reached it's write limitation and
+             // should be retried at a later time. We yield the processor and the FlowFile is considered unprocessed (partially processed) due to temporary write limitations.
+             // More about throughput limitations: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html
+            context.yield();
+            session.transfer(outgoingFlowFile, REL_UNPROCESSED);
+        } else if (result.getThrowable().getCause() instanceof AmazonClientException) {
+            getLogger().error("Could not process FlowFile due to client exception: " + result.getThrowable().getMessage(), result.getThrowable());
+            session.transfer(processClientException(session, Collections.singletonList(outgoingFlowFile), (AmazonClientException) result.getThrowable().getCause()), REL_FAILURE);
+        } else if (result.getThrowable().getCause() instanceof AmazonServiceException) {
+            getLogger().error("Could not process FlowFile due to server exception: " + result.getThrowable().getMessage(), result.getThrowable());
+            session.transfer(processServiceException(session, Collections.singletonList(outgoingFlowFile), (AmazonServiceException) result.getThrowable().getCause()), REL_FAILURE);
```

`AmazonServiceException` is a subclass of  `AmazonClientException` for some reason, so the "service" block should be moved before the "client" block. Otherwise the "client" branch would catch all exceptions.